pkg/generate/app/componentref.go

```diff
@@ -44,10 +44,6 @@ func componentWithSource(s string) (component, repo string, builder bool, err er
 	} else {
 		component = s
 	}
-	// component must be of the form compatible with a pull spec *or* <namespace>/<name>
-	if _, err := imageapi.ParseDockerImageReference(component); err != nil {
```

Comment:
How is this breaking a call like the one you mentioned? If we error out here, detection will continue to the next [check](https://github.com/openshift/origin/blob/18a92efac907eb5a5dcffa3c4f91fc3cd105f87f/pkg/generate/app/cmd/newapp.go#L146) which imho has to be corrected since this a template file specific problem.